hack/generate/samples/internal/go/v3/e2e_test_code.go

```diff
@@ -0,0 +1,426 @@
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// Run e2e tests using the Ginkgo runner.
+func TestE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+	fmt.Fprintf(GinkgoWriter, "Starting Memcached Operator suite\n")
+	RunSpecs(t, "Memcached e2e suite")
+}
+`
+
+const e2eTemplate = `/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	//nolint:golint
+	//nolint:revive
+	. "github.com/onsi/ginkgo"
+
+	//nolint:golint
+	//nolint:revive
+	. "github.com/onsi/gomega"
+
+	"github.com/example/memcached-operator/test/utils"
+)
+
+// namespace store the ns where the Operator and Operand will be executed
+const namespace = "memcached-operator-system"
+
+var _ = Describe("memcached", func() {
+
+	Context("ensure that Operator and Operand(s) can run in restricted namespaces", func() {
+		BeforeEach(func() {
+			By("installing prometheus operator")
+			Expect(utils.InstallPrometheusOperator()).To(Succeed())
+
+			By("installing the cert-manager")
+			Expect(utils.InstallCertManager()).To(Succeed())
+
+			By("creating manager namespace")
+			cmd := exec.Command("kubectl", "create", "ns", namespace)
+			_, _ = utils.Run(cmd)
+
+			By("labeling all namespaces to warn when we apply the manifest if would violate the PodStandards")
+			cmd = exec.Command("kubectl", "label", "--overwrite", "ns", "--all",
+				"pod-security.kubernetes.io/audit=restricted",
+				"pod-security.kubernetes.io/enforce-version=v1.24",
+				"pod-security.kubernetes.io/warn=restricted")
+			_, err := utils.Run(cmd)
+			ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+			By("labeling enforce the namespace where the Operator and Operand(s) will run")
+			cmd = exec.Command("kubectl", "label", "--overwrite", "ns", namespace,
+				"pod-security.kubernetes.io/audit=restricted",
+				"pod-security.kubernetes.io/enforce-version=v1.24",
+				"pod-security.kubernetes.io/enforce=restricted")
+			_, err = utils.Run(cmd)
+			Expect(err).To(Not(HaveOccurred()))
+		})
+
+		AfterEach(func() {
+			By("uninstalling the Prometheus manager bundle")
+			utils.UninstallPrometheusOperator()
+
+			By("uninstalling the cert-manager bundle")
+			utils.UninstallCertManager()
+
+			By("removing manager namespace")
+			cmd := exec.Command("kubectl", "create", "ns", namespace)
+			_, _ = utils.Run(cmd)
```

Comment:
Do we run `make uninstall` too? Envtest will clean up the cluster anyway by removing CRDs, but checking if we need that best practice in here too just so that users know.

Reply:
@varshaprasad96 we need to create the NS first to be able to label it as restricted. So that we can check if the manifest will or not be applied and are working under restricted enforcement.

Reply:
@varshaprasad96 this test is done done with ENVTEST. Why? 
ENVTEST does not allow us to do this e2e. It is more for you to test just the controller and not the whole solution.